src/common/utils.py

```diff
@@ -1,27 +1,28 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
-import sys
+from . import constants as CONSTANTS
 import json
 import copy
 import time
-from . import constants as CONSTANTS
-from . import debugger_communication_client
-from applicationinsights import TelemetryClient
+import sys
 
 previous_state = {}
 
 
-def show(state, debug_mode=False):
+def show(state, device_name, debug_mode=False):
     global previous_state
-    if state != previous_state:
-        previous_state = copy.deepcopy(state)
+
+    if not device_name in previous_state or state != previous_state[device_name]:
+
+        previous_state[device_name] = copy.deepcopy(state)
+        state_ext = {
+            "device_name": device_name,
+        }
+        state.update(state_ext)
```

Comment:
Can we put this outside of the if? If we can, I think it would streamline this method a bit.

Reply:
I put it all in a create_message helper function after fixing some stuff up (about to commit it). Does that help out? Do you mean streamline as in with clarity or runtime?

Reply:
I replied to this in the file changes area, but it doesn't seem to be here... :( here's what I said:
I put it all in a create_message helper function after fixing some stuff up (about to commit it). Does that help out? Do you mean streamline as in with clarity or runtime?

Reply:
Sorry, the comment was unclear! Chatted offline to elaborate what I meant.